### ISSUE TITLE
Scala.js packaging should produce NoModule by default

### ISSUE THREAD

**Issue by keynmol** [VirtusLab/scala-cli#338]
TL;DR: when packaging scala-cli and scalajs plugin have different defaults:

scala-cli defaults to CommonJS module type: https://github.com/VirtusLab/scala-cli/blob/a62e63e61b7b64084656246e5cada26a94a243ea/modules/build/src/main/scala/scala/build/options/ScalaJsOptions.scala#L37

Whereas Scala.js when used as SBT plugin defaults to NoModule: https://github.com/scala-js/scala-js/blob/9f7012281eda7185f64e9c3824cf316f2b761081/linker-interface/shared/src/main/scala/org/scalajs/linker/interface/StandardConfig.scala#L73

```scala
❯ cat chris.scala
// using scala 2.13.6
// using scala-js
import $dep.`org.scala-js::scalajs-dom::2.0.0`

import org.scalajs.dom
import org.scalajs.dom.document
import scala.scalajs.js.annotation.JSExportTopLevel

object Main extends App {
  def appendPar(targetNode: dom.Node, text: String): Unit = {
    val parNode = document.createElement("p")
    parNode.textContent = text
    targetNode.appendChild(parNode)
  }
  @JSExportTopLevel("addClickedMessage")
  def addClickedMessage(): Unit = {
    appendPar(document.body, "You clicked the button!")
  }
}
```
```
❯ scala-cli package chris.scala -f && (cat chris.js | grep addClickedMessage)
Compiling project (Scala 2.13.6, Scala.JS)
Compiled project (Scala 2.13.6, Scala.JS)
Wrote chris.js, run it with
  node ./chris.js
$c_LMain$.prototype.addClickedMessage__V = (function() {
exports.addClickedMessage = (function() {
  $m_LMain$().addClickedMessage__V()
```
```
❯ scala-cli package chris.scala --js-module-kind none -f && (cat chris.js | grep addClickedMessage)
Wrote chris.js, run it with
  node ./chris.js
let addClickedMessage;
$c_LMain$.prototype.addClickedMessage__V = (function() {
addClickedMessage = (function() {
  $m_LMain$().addClickedMessage__V()
```

Current default means you can't run the packaged script in the browser without altering configuration.

Stems from a [Discord discusion](https://discord.com/channels/632150470000902164/901153021638082660/904075644873342987)

**Comment by alexarchambault** [comment]
I think I went for common-js by default so that the generated files can be run straightaway with `node`. For the cases illustrated in the Scala CLI tests at least (like running munit tests).

As the failing tests in https://github.com/VirtusLab/scala-cli/pull/342 show, no-module creates issues when depending on munit for example, that seems to require file system-related APIs (which it gets via the `fs` module on node, so that some module support is required).

So I'm shared about making no-module the default. `--js-module-kind none` ought to work for the case described above.

**Comment by sjrd** [comment]
Isn't that an MUnit issue (https://github.com/scalameta/munit/issues/114, https://github.com/scalameta/munit/issues/247), which was fixed in https://github.com/scalameta/munit/pull/376, part of v0.7.27?

**Comment by ckipp01** [comment]
> I think I went for common-js by default so that the generated files can be run straightaway with node

I'm actually curious, what is the most common use case, running the created script with node, or trying to include it in a webpage? 

> So I'm shared about making no-module the default. --js-module-kind none ought to work for the case described above.

Gotcha, for a bit of context I think if someone is trying this out and reading through https://scala-cli.virtuslab.org/docs/guides/scala-js and trying to follow the ScalaJS docs will get tripped up on this. Since they'll probably think that they can just include the generated file in their `html` file like the ScalaJS docs talk about.

I have no issue passing `--js-module-kind none` in, but I do think beginners will trip up on this.

**Comment by sjrd** [comment]
> > I think I went for common-js by default so that the generated files can be run straightaway with node
> 
> I'm actually curious, what is the most common use case, running the created script with node, or trying to include it in a webpage?

The scripts generated for NoModule are also supposed to run straightaway with Node.js. This is how they are `run` and `test`ed by default with sbt-scalajs. ;) It was only an MUnit-specific issue that they didn't, which, as I mentioned above, I think is fixed.

**Comment by ckipp01** [comment]
> The scripts generated for NoModule are also supposed to run straightaway with Node.js. This is how they are run and tested by default with sbt-scalajs. ;) It was only an MUnit-specific issue that they didn't, which, as I mentioned above, I think is fixed.

Ahh ok, that's even better then.
